examples/fastapi/README.md

```diff
@@ -0,0 +1,126 @@
+# HumanLayer FastAPI Example
+
+This example demonstrates how to use AsyncHumanLayer with FastAPI to create an API with human-in-the-loop approval flows.
+
+## Overview
+
+The example shows how to:
+
+- Use AsyncHumanLayer with FastAPI endpoints
+- Handle human approvals for API operations
+- Process both approved and denied responses
+- Structure async API endpoints with human-in-the-loop flows
+
+## Setup
+
+1. Create a virtual environment and activate it:
+
+   ```bash
+   python -m venv venv
+   source venv/bin/activate  # On Windows: venv\Scripts\activate
+   ```
+
+2. Install dependencies:
+
+   ```bash
+   pip install -r requirements.txt
+   ```
+
+3. Copy `.env.example` to `.env` and fill in your API keys:
+   ```bash
+   cp .env.example .env
+   # Add your keys to .env:
+   # HUMANLAYER_API_KEY=your_key_here
+   # OPENAI_API_KEY=your_key_here
+   ```
+
+## Code Structure
+
+The main application (`app.py`) shows how to:
+
+1. Initialize AsyncHumanLayer:
+
+   ```python
+   from humanlayer import AsyncHumanLayer
+   hl = AsyncHumanLayer.cloud(verbose=True)
+   ```
+
+2. Create functions requiring approval:
+
+   ```python
+   @hl.require_approval()
+   async def multiply(a: int, b: int) -> int:
+       return a * b
+   ```
+
+3. Handle approvals in endpoints:
+   ```python
+   @app.post("/math/multiply")
+   async def math_multiply(a: int, b: int) -> Dict[str, str | int]:
+       result = await multiply(a, b)
+       if isinstance(result, str):  # Handle denial message
+           return {"status": "denied", "message": result}
+       return {"status": "success", "result": result}
+   ```
```

Comment:
I think since this is usage in deterministic code (function results are not being passed directly to the LLM) hl.fetch_approval() is probably more idiomatic.

```suggestion
or, with fetch_approval
   
   @app.post("/math/multiply")
   async def math_multiply(a: int, b: int) -> Dict[str, str | int]:
       approval_result = await hl.fetch_approval(
            FunctionCallSpec(
                fn="multiply", 
                kwargs={"a": a, "b": b},
            )
       )
       if not approval_result.approved:
           return {"status": "denied", "message": result}
       result = await multiply(a, b)
       return {"status": "success", "result": result}
```

This is more wordy but avoids the slightly weird `isinstance` call. We could document either or both ways - wdyt?